tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -0,0 +1,215 @@
+                as azure_upload_mock:
+            mock_credentials = ArtifactCredentialInfo(signed_uri=MOCK_AZURE_SIGNED_URI,
+                                                      type=ArtifactCredentialType.AZURE_SAS_URI)
+            write_credentials_response_proto = GetCredentialsForWrite.Response(
+                credentials=mock_credentials)
+            write_credentials_mock.return_value = write_credentials_response_proto
+            azure_upload_mock.return_value = None
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credentials_mock.assert_called_with(MOCK_RUN_ID, expected_location)
+            azure_upload_mock.assert_called_with(mock_credentials, test_file.strpath,
+                                                 expected_location)
+
+    @pytest.mark.parametrize("artifact_path,expected_location", [
+        (None, 'test.txt'),
+    ])
+    def test_log_artifact_aws(self, databricks_artifact_repo, test_file, artifact_path,
+                              expected_location):
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._get_write_credentials') \
+                as write_credentials_mock, \
+                mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._aws_upload_file') \
+                as aws_upload_mock:
+            mock_credentials = ArtifactCredentialInfo(signed_uri=MOCK_AWS_SIGNED_URI,
+                                                      type=ArtifactCredentialType.AWS_PRESIGNED_URL)
+            write_credentials_response_proto = GetCredentialsForWrite.Response(
+                credentials=mock_credentials)
+            write_credentials_mock.return_value = write_credentials_response_proto
+            aws_upload_mock.return_value = None
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credentials_mock.assert_called_with(MOCK_RUN_ID, expected_location)
+            aws_upload_mock.assert_called_with(mock_credentials, test_file.strpath)
+
+    def test_log_artifact_fail_case(self, databricks_artifact_repo, test_file, ):
+        mock_blob_service = mock.MagicMock(autospec=BlobClient)
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._get_write_credentials') \
+                as write_credentials_mock:
+            mock_credentials = ArtifactCredentialInfo(signed_uri=MOCK_AZURE_SIGNED_URI,
+                                                      type=ArtifactCredentialType.AZURE_SAS_URI)
+            write_credentials_response_proto = GetCredentialsForWrite.Response(
+                credentials=mock_credentials)
+            write_credentials_mock.return_value = write_credentials_response_proto
+            mock_blob_service.from_blob_url().return_value = MlflowException("MOCK ERROR")
+            with pytest.raises(MlflowException):
+                databricks_artifact_repo.log_artifact(test_file.strpath)
+                write_credentials_mock.assert_called_with(MOCK_RUN_ID, ANY)
```

Comment:
I don't think this `assert` function will be called since the `pytest.raises` scope terminates as soon as an exception is encountered. You should be able to fix this by moving the `assert` call out of the `pytest.raises` context (i.e., unindent 1 level).

Reply:
Done.